### PR TITLE
Support for aarch64-apple-darwin

### PR DESCRIPTION
**Pull request by matthewtgilbride**

### CODE REVIEW

**Comment by brooksmtownsend** [comment]
@ChrisRx do we support `aarch64-apple-darwin` off of your fork of `cross`? I suppose related question @matthewtgilbride, were you able to verify that this worked by building with the cross image? I can try on my computer once I'm on for the day to give it a shot.

I think you're correct here though, this is what we'll want to do to support the M1 machines for all capability providers 😄

**Comment by matthewtgilbride** [comment]
@brooksmtownsend I created the PR and immediately grabbed my work (non-M1) laptop assuming no one was up yet 😆 

I'll give it a try at lunch and report back.

**Comment by matthewtgilbride** [comment]
@brooksmtownsend I run into local gcc issues trying to build the whole project from M1.  Was about to try and build from an x86 machine but you mention @ChrisRx 's fork of cross, which I assume is this? https://github.com/ChrisRx/cross

**Comment by brooksmtownsend** [comment]
I think the main thing we're missing here is the `aarch64-apple-darwin` docker image, cross works by running the `cargo build` step in a container that matches that arch/os pair, and the `x86_64-apple-darwin` part works because that image exists (thanks to @ChrisRx). We may need to add in that container ourselves, unfortunately the cross maintainers haven't merged in that PR to the main branch yet

**Comment by brooksmtownsend** [comment]
@matthewtgilbride Quick question, to clarify, are you able to build `http-server` with a straight up `cargo build` on the M1?

**Review by autodidaddict — APPROVED** [review]

File: Cross.toml
Comment (brooksmtownsend) [inline Cross.toml:26]: We're going to want to update the `aarch64-apple-darwin` image to something that actually exists, and we're going to have to verify it works before we merge this in. Currently, the `criticalstack/cross:aarch64-apple-darwin` image doesn't exist and we need to find out if we can actually create that image

File: http-server/Cross.toml
Comment (brooksmtownsend) [inline http-server/Cross.toml:26]: Same comment as above

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]

**Comment by brooksmtownsend** [comment]
Because @ChrisRx has done the hard part with https://github.com/ChrisRx/cross-images, I suggest we just fork that to the wasmcloud repository and update the images accordingly. For now we can just update the x86_64-linux one as that's the only image giving us issues. Thoughts @autodidaddict ?

**Comment by autodidaddict** [comment]
Make it so

**Comment by autodidaddict** [comment]
What's the status on this?

**Comment by brooksmtownsend** [comment]
Looks like my previous comment was in reference to a different issue. For this, we'll need to:

- [ ] Fork criticalstack/cross to wasmcloud repo
- [ ] duplicate https://github.com/ChrisRx/cross-images/blob/main/Dockerfile.x86_64-apple-darwin and modify for `aarch64-apple-darwin`
- [ ] Ensure the above works
- [ ] Release our images to our dockerhub.

I haven't performed any of these steps, but I'll get started on steps 1+2. Will likely need @matthewtgilbride to test running the parJEEzy afterwards

**Comment by brooksmtownsend** [comment]
I'm not 100% sure that this will work, https://github.com/tpoechtrager/osxcross#what-is-the-goal-of-osxcross states that you can target `aarch64` with it so in theory we should be able to provide this.

**Comment by autodidaddict** [comment]
@brooksmtownsend status on this?

**Comment by brooksmtownsend** [comment]
I've got the docker images ready, was planning on putting a PR into @matthewtgilbride 's fork today once I can verify that it builds well 👍🏻 (and once @matthewtgilbride can verify that the parJEEzy works)

**Comment by brooksmtownsend** [comment]
RE: my comment yesterday, @matthewtgilbride was able to confirm that the aarch64-apple-darwin is working for him 🚀 Barring an update to the blobstore contract, I've submitted https://github.com/matthewtgilbride/capability-providers/pull/3, which will get merged in, and then we can merge this into main.

**Review by brooksmtownsend — APPROVED** [review]
Once tests pass, LGTM 👍🏻

**Pull request merged**
